examples/test_examples.py

```diff
@@ -50,16 +49,101 @@ def flaky_print(*args):
         frontends[frontend] = True
     except ImportError:
         pass
+    except ModuleNotFoundError:
```

Comment:
ˋModuleNotFoundErrorˋ is a subclass of ˋImportErrorˋ, so excepting ˋImportErrorˋ should also catch ˋModuleNotFoundErrorˋ. More problematic, ˋModuleNotFoundErrorˋ is not available in Python 2.7, so this will actually raise a NameError for ModuleNotFoundError once an exception is handled.
It does not fail currently because Python is so dynamic it would allow you to define the class ModuleNotFoundError inside the try block and excepting it then directly after.

Reply:
Wasn't sure if catching `ImportError` would catch `ModuleNotFoundErrors`s or not...regardless, each enough to remove if the `ImportError` is sufficient to catch it.

Reply:
Have to admit that I also stumbled across this when making the hdf5 PRs, but turns out that ModuleNotFoundError is a fancier ImportError (it has an attribute "name" containing the name of the module that was not found): https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError

Reply:
be0e95a incorporates this change